extensions/ql-vscode/src/debugger/debug-session.ts

```diff
@@ -300,13 +437,25 @@ export class QLDebugSession extends LoggingDebugSession implements Disposable {
       this.sendEvent(outputEvent);
     }
 
+    if (this.terminateOnComplete) {
+      this.terminateAndExit();
+    } else {
+      // Report the session as "stopped", but keep the session open.
+      this.sendEvent(new StoppedEvent("entry", QUERY_THREAD_ID));
+
+      this.state = "stopped";
+    }
+    this.queryRun = undefined;
+  }
+
+  private terminateAndExit(): void {
     // Report the debugging session as terminated.
     this.sendEvent(new TerminatedEvent());
 
     // Report the debuggee as exited.
-    this.sendEvent(new ExitedEvent(result.resultType));
+    this.sendEvent(new ExitedEvent(this.lastResult!.resultType));
```

Comment:
I guess here, `!` is unavoidable.

Reply:
I realized that I was only saving `lastResult` for the exit code (`resultType`), so I changed it to just save `resultType` with a default value.